### DISCUSSION TITLE
How to create other users without Auth SingUp process.

### DISCUSSION THREAD

**Discussion by DefMaks** [supabase/supabase#1327] [Feature Requests]
I'm currently building a dashboard based on user privileges or roles. So i'd like only one signup (the super admin, which is done perfectly). This one should be able to create other users via some forms so they can either receive or not the account details in the mail early provided. But i'm stuck.
Any solutions please ?

**Comment by silentworks** [comment]
This is not documented but I'm almost certain it's what Supabase is using in their dashboard when you invite a new user.

```js
await supabase.auth.api.inviteUserByEmail('email_address_here')
```

**Reply by silentworks** [reply]
To expand on what I mentioned above, when an invite is sent a new user is created in the `auth.users` table, you can follow the instructions on creating a `public.users` table which keeps a reference to `auth.users` to store user-specific information. When you use the `await supabase.auth.api.inviteUserByEmail` it returns some data which should contain the `id` for the user. You can use this to fill out all the missing specific user information you may already have or wait for the user to click the invite link and verify themselves before asking for the additional information.

**Reply by DefMaks** [reply]
Good spot, i'll try it out,
Thanks a lot @silentworks

**Reply by DefMaks** [reply]
Hi @silentworks , thanks a lot for the tip, and sorry for the late reply.
Yeah finally did it and it's ok for a first level. I wanted to invite first then set some custom informations. 
Ex: a md5 Default password and a custom user_meta for the new created user.

I read the doc but i cannot edit another user's data. 
Is there any way to make it?

**Reply by silentworks** [reply]
If you use the Supabase API from the server-side and use the `service_role` `secret` you can accomplish this.

**Reply by AlexNeep** [reply]
The supabase request was failing for the `inviteUserByEmail` method.

But I got it working with a work-around. You can create the user on a backend server using the normal `supabase.auth.signUp({email, password})` method.

This will leave the client logged in but still create a new account for the new user

**Reply by T0nyBoy** [reply]
hi Alex. As I am new with Supabase and do not get it. What do you mean by server? I have the exact same problem but as I am using React, there is no backend. Please could you explain so I may find a solution? Thanks :)

**Reply by AlexNeep** [reply]
Hey, no worries. React runs in the browser. You create a react application that is delivered and rendered on the browser which you then see when you view your app. 

If I recall correctly, the problem was that if you run `signUp` within the browser, Supabase with also logs in that user into the supabase session with their new account. This is not helpful if you are creating an account for another user as it will log out the original user. It probably does this by leaving a session token in the browser which identifys the new user.

To get around this, you can also run this code on a server somewhere (The server can be hosted anywhere). 

When we use a server like this it is often called an API. So you will just have to send an API request using something like `axios` to that API/server. The server/API will then process the request and sign up the supabase user. But as this doesn't happen in the browser, the logged in user does not change.

You can create your API/server anywhere. If you use react with next or remix, then you might find it easier to setup: [see here](https://nextjs.org/docs/api-routes/introduction)

**Reply by T0nyBoy** [reply]
Hi Alex. Thank you very much! I totally understand what you are describing now. :)

**Comment by MohammedHafiz** [comment]
> 

Hey @silentworks, Can you please explain how to do this on the server side using secret_key

**Reply by johndpope** [reply]
https://supabase.com/docs/reference/javascript/supabase-auth-admin-api

https://supabase.com/docs/reference/javascript/auth-admin-createuser

```javascript

const { data, error } = await supabase.auth.admin.createUser({
  email: 'user@email.com',
  password: 'password',
  user_metadata: { name: 'Yoda' }
})
```
